crates/type-system/Cargo.toml

```diff
@@ -0,0 +1,40 @@
+[package]
+name = "type-system"
+version = "0.0.0"
+edition = "2021"
+authors = ["HASH"]
+publish = false
+description = "Definitions of types within the Block Protocol Type System"
+
+[lib]
+name = "type_system"
+crate-type = ["cdylib", "rlib"]
+
+[features]
+default = ["console_error_panic_hook"]
+
+[dependencies]
+serde = { version = "1.0.137", features = ["derive"] }
+thiserror = "1.0.32"
+url = "2.2.2"
```

Comment:
Turns out URL is adding a _**lot**_ (~300-400kb) to the size of the WASM binary. This is a [known issue](https://github.com/servo/rust-url/issues/557) but I don't think they're planning on addressing it.

My current thinking is we go with it until it seems like the size is a problem, at which point we could write a lightweight URL thing that does what we need from it, or we could conditionally [utilise the browser's implementation](https://docs.rs/web-sys/0.3.45/web_sys/struct.Url.html) of `url` in a wasm environment.

Either way we have options and I don't think we need to worry right now.